sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -4309,6 +4362,98 @@ protected internal virtual BlobContainerClient GetParentBlobContainerClientCore(
             return _parentBlobContainerClient;
         }
         #endregion
+
+        #region GetVersions
+        /// <summary>
+        /// Gets the Blob Versions associated with the Blob.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// An <see cref="Pageable{T}"/> of <see cref="BlobItem"/>
+        /// describing the versions assocaited with the blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Pageable<BlobItem> GetBlobVersions(CancellationToken cancellationToken = default)
+            => ContainerClient.GetBlobs(
+                states: BlobStates.Version,
+                prefix: Name,
+                cancellationToken: cancellationToken);
+
+        /// <summary>
+        /// Gets the Blob Versions associated with the Blob.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// An <see cref="Pageable{T}"/> of <see cref="BlobItem"/>
+        /// describing the versions assocaited with the blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual AsyncPageable<BlobItem> GetBlobVersionsAsync(CancellationToken cancellationToken = default)
+            => ContainerClient.GetBlobsAsync(
+                states: BlobStates.Version,
+                prefix: Name,
+                cancellationToken: cancellationToken);
+        #endregion
+
+        #region GetSnapshots
+        /// <summary>
+        /// Gets the Blob Snaphots associated with the Blob.
+        /// Note that the current version of the Blob will be returned
```

Comment:
This method returns [number of snapshots] + 1 `BlobItems`.  This is because the service also returns the current Blob.